### DISCUSSION TITLE
A UI library that uses fp-ts and JSX

### DISCUSSION THREAD

**Discussion by ej-shafran** [gcanti/fp-ts#1904] [Show and tell]
Hi y'all!

I had a bunch of free time and a desire to do something Elm-like using `fp-ts`, so I made myself a little library and thought I'd share!

There's [the library itself](https://www.npmjs.com/package/ui-ts) and a [CLI tool for starting new project](https://www.npmjs.com/package/ui-ts-cli).

It's pretty basic right now, but I found it fun to play around and build stuff with :smile_cat:

In the future I plan to add types to the JSX so it's not just the wild west, and any feature requests are more than welcome :)

**Comment by ElijahKotyluk** [comment]
Going to check this out, thank you for putting it together.

**Comment by ej-shafran** [comment]
Happy to inform that I have, in fact, added types to the JSX :)
So while still incomplete it's much nicer to code with now
